src/components/menu/Menu.js

```diff
@@ -0,0 +1,105 @@
+//Libraries
+import React from 'react';
+import PropTypes from 'prop-types';
+
+//Helpers
+import { getDishById } from './helpers/menuHelper';
+import _omit from 'lodash/omit';
+
+//Constants
+import { INITIAL_STATE, CHECKOUT } from './constants/menu.general';
+
+//Components
+import MenuLinks from './components/menuLinks';
+import Dishes from './components/dishes';
+import Cart from './components/cart';
+
+//CSS
+import './menu.css';
+
+class Menu extends React.Component {
+  state = INITIAL_STATE;
+
+  resetCartItems = () => {
+    this.setState({
+      cartItems: [],
+      quantities: {},
+    });
+  };
+
+  handleEvents = (dishId = CHECKOUT, changeInQuantity = 1) => {
+    if (dishId === CHECKOUT) {
+      this.handleCheckout();
+    } else {
+      this.handleCartChange(dishId, changeInQuantity);
+    }
+  };
+
+  handleCheckout = () => {
+    const { cartItems } = this.state;
+    localStorage.setItem('Cart Items', JSON.stringify(cartItems));
+    this.resetCartItems();
+  };
+
+  handleCartChange = (dishId, changeInQuantity) => {
```

Comment:
try to update this function once using immutability principle as prakhar suggested